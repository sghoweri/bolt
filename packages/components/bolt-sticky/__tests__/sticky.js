import {
  render,
  renderString,
  stopServer,
} from '../../../../packages-extras/testing/testing-helpers';

describe('<bolt-sticky> Component', () => {
  test('basic usage', async () => {
    afterAll(async () => {
      await stopServer();
    });

    const results = await renderString(
      `
      <div style="height:50vh;background-color:lightgray;"></div>

      {% embed "@bolt-components-sticky/sticky.twig" %}
        {% block sticky_content %}
          <div style="color:white;background-color:red">
            First sticky item
          </div>
        {% endblock %}
      {% endembed %}

      <div style="height:100vh;background-color:lightgray;"></div>

      {% embed "@bolt-components-sticky/sticky.twig" %}
        {% block sticky_content %}
          <div style="background-color: white">
            Second sticky item
          </div>
        {% endblock %}
      {% endembed %}

      <div style="height:100vh;background-color:lightgray;"></div>

      {% embed "@bolt-components-sticky/sticky.twig" %}
        {% block sticky_content %}
          <div style="color:white;background-color:blue">
            Third sticky item
          </div>
        {% endblock %}
      {% endembed %}

      <div style="height:100vh;background-color:lightgray;"></div>
        `,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
  test('usage with content', async () => {
    const results = await renderString(
      `
      {% embed "@bolt-components-sticky/sticky.twig" %}
      {% block sticky_content %}
        {% include "@bolt-components-navbar/navbar.twig" with {
          "theme": "dark",
          "title":
          {
            "tag": "h2",
            "text": "Pega for Government: Healthcare & Social Programs",
            "icon":
            {
              "name": "marketing-gray"
            }
          },
          "links":
          [
            {
              "text": "Visualization & Simulation",
              "url": "#visualization-and-simulation"
            },
            {
              "text": "Section 1",
              "url": "#section-1"
            },
            {
              "text": "Section 2 is a lot longer",
              "url": "#section-2"
            },
            {
              "text": "The constituent experience",
              "url": "#the-constituent-experience"
            },
            {
              "text": "Section 4",
              "url": "#section-4"
            }
          ]
        } only %}
      {% endblock %}
    {% endembed %}

    {% embed "@bolt-components-band/band.twig" with {
      size: "large",
      theme: "dark",
      full_bleed: true,
      attributes: {
        id: [
          "section-2"
        ]
      }
    } %}
      {% block band_content %}

        {% include "@bolt-components-headline/headline.twig" with {
          text: "Section 2",
          size: "xxlarge",
          tag: "h2"
        } only %}

        {% include "@bolt-components-headline/text.twig" with {
          text: "Suspendisse dictum feugiat nisl ut dapibus. Mauris iaculis porttitor posuere. Praesent id metus massa, ut blandit odio. Proin quis tortor orci. Etiam at risus et justo dignissim congue. Donec congue lacinia dui, a porttitor lectus condimentum laoreet. Nunc eu ullamcorper orci. Quisque eget odio ac lectus vestibulum faucibus eget in metus. In pellentesque faucibus vestibulum. Nulla at nulla justo, eget luctus tortor. Nulla facilisi. Duis aliquet egestas purus in blandit. Curabitur vulputate, ligula lacinia scelerisque tempor, lacus lacus ornare ante, ac egestas est urna sit amet arcu. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed molestie augue sit amet leo consequat posuere. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Proin vel ante a orci tempus eleifend ut et magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus luctus urna sed urna ultricies ac tempor dui sagittis. In condimentum facilisis porta. Sed nec diam eu diam mattis viverra. Nulla fringilla, orci ac euismod semper, magna diam porttitor mauris, quis sollicitudin sapien justo in libero. Vestibulum mollis mauris enim. Morbi euismod magna ac lorem rutrum elementum. Donec viverra auctor lobortis. Pellentesque eu est a nulla placerat dignissim. Morbi a enim in magna semper bibendum. Etiam scelerisque, nunc ac egestas consequat, odio nibh euismod nulla, eget auctor orci nibh vel nisi. Aliquam erat volutpat. Mauris vel neque sit amet nunc gravida congue sed sit amet purus. Quisque lacus quam, egestas ac tincidunt a, lacinia vel velit. Aenean facilisis nulla vitae urna tincidunt congue sed."
        } only %}

        {% include "@bolt-components-headline/text.twig" with {
          text: "Quisque eget odio ac lectus vestibulum faucibus eget in metus. In pellentesque faucibus vestibulum. Nulla at nulla justo, eget luctus tortor. Nulla facilisi. Duis aliquet egestas purus in blandit. Curabitur vulputate, ligula lacinia scelerisque tempor, lacus lacus ornare ante, ac egestas est urna sit amet arcu. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed molestie augue sit amet leo consequat posuere. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Proin vel ante a orci."
        } only %}

      {% endblock band_content %}
    {% endembed %}
        `,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
