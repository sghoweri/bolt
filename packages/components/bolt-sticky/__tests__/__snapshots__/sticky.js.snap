// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-sticky> Component basic usage 1`] = `
<div style="height:50vh;background-color:lightgray;">
</div>
<bolt-sticky bolt-component
             class="c-bolt-sticky js-bolt-sticky"
>
  <div style="color:white;background-color:red">
    First sticky item
  </div>
</bolt-sticky>
<div style="height:100vh;background-color:lightgray;">
</div>
<bolt-sticky bolt-component
             class="c-bolt-sticky js-bolt-sticky"
>
  <div style="background-color: white">
    Second sticky item
  </div>
</bolt-sticky>
<div style="height:100vh;background-color:lightgray;">
</div>
<bolt-sticky bolt-component
             class="c-bolt-sticky js-bolt-sticky"
>
  <div style="color:white;background-color:blue">
    Third sticky item
  </div>
</bolt-sticky>
<div style="height:100vh;background-color:lightgray;">
</div>
`;

exports[`<bolt-sticky> Component usage with content 1`] = `
<bolt-sticky bolt-component
             class="c-bolt-sticky js-bolt-sticky"
>
  <bolt-navbar class="t-bolt-dark"
               width="full"
  >
    <nav class="c-bolt-navbar c-bolt-navbar--full ">
      <div class="c-bolt-navbar__title">
        <div class="c-bolt-navbar__title--static">
          <div class="c-bolt-navbar__title-icon">
            <bolt-icon name="marketing-gray"
                       size="large"
            >
              <span class="c-bolt-icon c-bolt-icon--marketing-gray c-bolt-icon--large">
                <svg class="c-bolt-icon__icon c-bolt-icon__icon--large"
                     xmlns="http://www.w3.org/2000/svg"
                     data-name="Layer 1"
                     viewbox="0 0 32 32"
                >
                  <path fill="var(--bolt-theme-icon, currentColor)"
                        fill-rule="evenodd"
                        d="M30.3 8.8H28a13.7 13.7 0 012 7.2A14 14 0 1116 2a14.5 14.5 0 017.4 2.1V1.8A15.3 15.3 0 0016 0a16 16 0 1016 16 15.6 15.6 0 00-1.7-7.2"
                        data-name="Fill-1"
                  >
                  </path>
                  <path fill="var(--bolt-theme-icon, currentColor)"
                        fill-rule="evenodd"
                        d="M16 8a8.2 8.2 0 013.1.6l1.5-1.5A10.4 10.4 0 0016 6a10 10 0 1010 10 9.4 9.4 0 00-1.2-4.7l-1.5 1.5A8.8 8.8 0 0124 16a8 8 0 11-8-8"
                        data-name="Fill-4"
                  >
                  </path>
                  <path fill="var(--bolt-theme-icon, currentColor)"
                        fill-rule="evenodd"
                        d="M16 16.5a.5.5 0 11.5-.5.5.5 0 01-.5.5zm.9-2.8a3 3 0 00-.9-.2 2.5 2.5 0 102.5 2.5 2.4 2.4 0 00-.2-.9L25.4 8H30V6h-2.6l1.8-1.8-1.4-1.4L26 4.5V2h-2v4.6z"
                        data-name="Fill-6"
                  >
                  </path>
                </svg>
              </span>
            </bolt-icon>
          </div>
          <h2 class="c-bolt-navbar__title-text">
            Pega for Government: Healthcare & Social Programs
          </h2>
        </div>
      </div>
      <bolt-nav-priority more-text="More">
        <nav class="c-bolt-nav-priority">
          <bolt-nav-indicator>
            <ul class="c-bolt-nav-priority__list c-bolt-nav-priority__primary">
              <li class="c-bolt-nav-priority__item">
                <bolt-navlink>
                  <a href="#visualization-and-simulation"
                     class="c-bolt-navlink"
                  >
                    <span class="c-bolt-navlink__text">
                      Visualization & Simulation
                    </span>
                  </a>
                </bolt-navlink>
              </li>
              <li class="c-bolt-nav-priority__item">
                <bolt-navlink>
                  <a href="#section-1"
                     class="c-bolt-navlink"
                  >
                    <span class="c-bolt-navlink__text">
                      Section 1
                    </span>
                  </a>
                </bolt-navlink>
              </li>
              <li class="c-bolt-nav-priority__item">
                <bolt-navlink>
                  <a href="#section-2"
                     class="c-bolt-navlink"
                  >
                    <span class="c-bolt-navlink__text">
                      Section 2 is a lot longer
                    </span>
                  </a>
                </bolt-navlink>
              </li>
              <li class="c-bolt-nav-priority__item">
                <bolt-navlink>
                  <a href="#the-constituent-experience"
                     class="c-bolt-navlink"
                  >
                    <span class="c-bolt-navlink__text">
                      The constituent experience
                    </span>
                  </a>
                </bolt-navlink>
              </li>
              <li class="c-bolt-nav-priority__item">
                <bolt-navlink>
                  <a href="#section-4"
                     class="c-bolt-navlink"
                  >
                    <span class="c-bolt-navlink__text">
                      Section 4
                    </span>
                  </a>
                </bolt-navlink>
              </li>
            </ul>
          </bolt-nav-indicator>
        </nav>
      </bolt-nav-priority>
    </nav>
  </bolt-navbar>
</bolt-sticky>
<bolt-band tag="div"
           size="large"
           theme="dark"
           valign="middle"
           full-bleed
           id="section-2"
>
  <div class="c-bolt-band c-bolt-band--large c-bolt-band--valign-middle t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          <h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxlarge">
            Section 2
          </h2>
          <p class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--medium">
            Suspendisse dictum feugiat nisl ut dapibus. Mauris iaculis porttitor posuere. Praesent id metus massa, ut blandit odio. Proin quis tortor orci. Etiam at risus et justo dignissim congue. Donec congue lacinia dui, a porttitor lectus condimentum laoreet. Nunc eu ullamcorper orci. Quisque eget odio ac lectus vestibulum faucibus eget in metus. In pellentesque faucibus vestibulum. Nulla at nulla justo, eget luctus tortor. Nulla facilisi. Duis aliquet egestas purus in blandit. Curabitur vulputate, ligula lacinia scelerisque tempor, lacus lacus ornare ante, ac egestas est urna sit amet arcu. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed molestie augue sit amet leo consequat posuere. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Proin vel ante a orci tempus eleifend ut et magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus luctus urna sed urna ultricies ac tempor dui sagittis. In condimentum facilisis porta. Sed nec diam eu diam mattis viverra. Nulla fringilla, orci ac euismod semper, magna diam porttitor mauris, quis sollicitudin sapien justo in libero. Vestibulum mollis mauris enim. Morbi euismod magna ac lorem rutrum elementum. Donec viverra auctor lobortis. Pellentesque eu est a nulla placerat dignissim. Morbi a enim in magna semper bibendum. Etiam scelerisque, nunc ac egestas consequat, odio nibh euismod nulla, eget auctor orci nibh vel nisi. Aliquam erat volutpat. Mauris vel neque sit amet nunc gravida congue sed sit amet purus. Quisque lacus quam, egestas ac tincidunt a, lacinia vel velit. Aenean facilisis nulla vitae urna tincidunt congue sed.
          </p>
          <p class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--medium">
            Quisque eget odio ac lectus vestibulum faucibus eget in metus. In pellentesque faucibus vestibulum. Nulla at nulla justo, eget luctus tortor. Nulla facilisi. Duis aliquet egestas purus in blandit. Curabitur vulputate, ligula lacinia scelerisque tempor, lacus lacus ornare ante, ac egestas est urna sit amet arcu. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed molestie augue sit amet leo consequat posuere. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Proin vel ante a orci.
          </p>
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>
`;
